src/spec.c

```diff
@@ -1366,6 +1369,13 @@ static int FieldSpec_RdbLoad(RedisModuleIO *rdb, FieldSpec *f, int encver) {
     f->tagSep = *s;
     RedisModule_Free(s);
   }
+  // Load vector specific options
+  if (encver >= INDEX_VECSIM_VERSION ||
+      FIELD_IS(f, INDEXFLD_T_VECTOR) || (f->options & FieldSpec_Dynamic)) {
```

Comment:
should it be the following?
```suggestion
  if (encver >= INDEX_VECSIM_VERSION &&
      (FIELD_IS(f, INDEXFLD_T_VECTOR) || (f->options & FieldSpec_Dynamic))) {
```